gameMechanics.js

```diff
@@ -0,0 +1,303 @@
+	var sum = 0;
+	var currentSumHorizontal = 1 + checkLeftHoriz(currentGrid, selectedColumn, currentPosition, currentPlayer, sum) + checkRightHoriz(currentGrid, selectedColumn, currentPosition, currentPlayer, sum);
+	var currentSumMainDiagonal = 1 + checkUpperLeftDiag(currentGrid, selectedColumn, currentPosition, currentPlayer, sum) + checkLowerRightDiag(currentGrid, selectedColumn, currentPosition, currentPlayer, sum);
+    var currentSumAntiDiagonal = 1 + checkLowerLeftDiag(currentGrid, selectedColumn, currentPosition, currentPlayer, sum) + checkUpperRightDiag(currentGrid, selectedColumn, currentPosition, currentPlayer, sum);
+	var currentSumVertical = 1 + checkVertical(currentGrid, selectedColumn, currentPosition, currentPlayer, sum);
+	if (currentSumHorizontal >= 4 || currentSumMainDiagonal >= 4 || currentSumAntiDiagonal >= 4 || currentSumVertical >= 4) {
+		if (currentPlayer % 2 == 0) {
+			alert ("Player Two Wins!");
+			scorePlayerTwo++;
+		}
+		else {
+			alert ("Player One Wins!");
+			scorePlayerOne++;
+		}
+		changeButtons();
+		printScore(scorePlayerOne, scorePlayerTwo);
+	}
+}
+
+var resetGame = function(currentGrid) {
+	var height = currentGrid[0].length;
+    var width = currentGrid.length;
+	mainGrid = createGrid(height,width);
+	printGrid(mainGrid, "mainGrid");
+	turn = 1;
+	printTurn(turn);
+	document.getElementById("buttonA").disabled = false;
+	document.getElementById("buttonB").disabled = false;
+	document.getElementById("buttonC").disabled = false;
+	document.getElementById("buttonD").disabled = false;
+	document.getElementById("buttonE").disabled = false;
+	document.getElementById("buttonF").disabled = false;
+	document.getElementById("buttonG").disabled = false;
+	document.getElementById("reset").value = "Start Over";	
+}
+
+var changeButtons = function() {
+	document.getElementById("buttonA").disabled = true;
+	document.getElementById("buttonB").disabled = true;
+	document.getElementById("buttonC").disabled = true;
+	document.getElementById("buttonD").disabled = true;
+	document.getElementById("buttonE").disabled = true;
+	document.getElementById("buttonF").disabled = true;
+	document.getElementById("buttonG").disabled = true;
+	document.getElementById("reset").value = "Play Again!";	
+}
+
+var printScore = function(scorePlayerOne, scorePlayerTwo) {
+	document.getElementById("scoreBoard").innerHTML = "<b>Score Board:</b><br>Player One:" + " " + scorePlayerOne + "<br><font color='red'>Player Two:" + " " + scorePlayerTwo + "</font><br>";
+}
+
+var printTurn = function (turn) {
+	var declareTurn = "Player One (X)";
+	if (turn % 2 == 0) {
+		declareTurn = "<font color='red'>Player Two (O)</font>";
+	}
+	document.getElementById("currentTurn").innerHTML = "Current Turn:" + " " + declareTurn + " ";
+}
+
+var checkLeftHoriz = function(currentGrid, selectedColumn, currentRow, currentPlayer, currentSum) {
+	var newSum = currentSum;
+	if (selectedColumn == 0) return currentSum;
```

Comment:
You may want to consider always using brackets. It's explicit and helps you avoid potentially bad bugs from accidentally including/excluding things inside the `if` condition. There's a famous story floating around the internet somewhere... think it was Apple or some other big company.